lib/chatty/connection.ex

```diff
@@ -119,10 +144,24 @@ defmodule Chatty.Connection do
     {:noreply, %{state | sock: nil, last_message_time: nil}}
   end
 
+  def handle_info({:ssl_error, sock, reason}, %{sock: sock} = state) do
```

Comment:
Right, each `handle_info` corresponds to a different message that we can receive. However, both `handle_info({:ssl_error, ...})` and `handle_info({:tcp_error, ...})` should call the same local function. The reason for that is that we have a non-trivial behaviour that we need to perform when there's an error:

1. Close the socket.
2. Call reconnect.
3. Update the state.

If we have those steps duplicated in multiple places then in the future when someone changes the behaviour in one place, they are more likely to forget or mess up applying the same change to all the other ones.

Moreover, the last two steps are also performed in `handle_info({:tcp_closed, ...})`. I'm responsible for introducing that duplication initially, but now have a great chance to clean it all up.